kubernetes-gcp-yaml/Pulumi.yaml

```diff
@@ -15,10 +15,9 @@ template:
 
 config:
   gcp:project:
-    type: string
+    value: changeme
   gcp:region:
-    type: string
-    default: us-central1
+    value: us-central1
```

Comment:
I _think_ the fix for this is to remove these provider config settings from the root `config` block, as they'll be set during `pulumi new` using defaults from `template.config`. When I run `pulumi new` with this branch, for example, I don't see `changeme` in the prompt as a default, and after creation I end up with an empty string in `Pulumi.dev.yaml`: 

```yaml
config:
  asasdasdfa:nodesPerZone: "1"
  gcp:project: ""
  gcp:region: us-central1
```

Tried with 3.48 and 3.49 and got the same result. Looking at other `*-gcp-yaml` templates, it seems they all leave these it out of the program config sections, so we should be good just removing it.

Reply:
We can't leave those out; they are referenced later in the template. In order to be able to reference them later, they have to be included in the `config` block. And there has to be at least a value key under each of them, because the template won't work otherwise.

Reply:
I see, ok. We’ll if this fixes it, let’s go ahead and merge — we can always come back it. Thanks for the explanation and the fix!

Reply:
cc @aq17 @AaronFriel Curious what we would expect to be used here?  It doesn't feel right to require that we include dummy values like this?

Reply:
Yes, for keys that should have a default, using `value` instead of `type` and `default` is correct, but using a dummy seems like it could lead to confusion. 
We could consider allowing project config keys to just be declared and not require a value within the `Pulumi.yaml` file?

Reply:
Actually, it is possible to omit `gcp:project` and `gcp:region` from the template and still reference them; when running `pulumi up`, you should get 
```
    warning: unable to detect a global setting for GCP Project;
    Pulumi will rely on per-resource settings for this operation.
    Set the GCP Project by using:
    	`pulumi config set gcp:project <project>` 
```
if it isn't set in your stack config, which seems preferable to using dummy values.

Reply:
@aq17 Interesting; I was under the impression that the only way to use a configuration value (provider-namespaced or otherwise) was to include it in the `config` block (the template does reference `gcp:project` later on). I'll try removing it from the `config` block and report back.

Reply:
> I think the fix for this is to remove these provider config settings from the root config block

> Actually, it is possible to omit gcp:project and gcp:region from the template and still reference them

Finally got a chance to try this locally myself, and just removing those two settings does seem to do the trick, yeah.

We should also remove them from the template's page on pulumi.com: https://www.pulumi.com/templates/kubernetes/gcp/#customizing-the-project

Reply:
I also tested this locally and it seemed to work; I wanted to run one more set of tests to be sure. Agreed that if we remove these settings we need to update the landing page as well.

Reply:
I've run a few tests and can't find any ill effects from removing `gcp:project` and `gcp:region` from the top-level `config` block, so I'll push another commit that just removes those and re-request reviews before merging.